### ISSUE TITLE
case class TransactionRequestBody is not named correctly

### ISSUE THREAD

**Issue by simonredfern** [OpenBankProject/OBP-API#124]
The case class TransactionRequestBody is not really named correctly. - A better name for it would be TransactionRequestBodySandboxTan since this is what it currently represents.

So perhaps we should make it a subclass something like: 

  abstract class TransactionRequestBody {
    val value : AmountOfMoney
  }

  case class TransactionRequestBodySandBoxTan(
    val to: TransactionRequestAccount,
    val description : String
  )  extends TransactionRequestBody

however at first glance, this doesn't compile - and doesn't feel right in some of the situations it is used.

**Comment by simonredfern** [comment]
e.g. in my locally modified code i have:   override def createTransactionRequestImpl(transactionRequestId: TransactionRequestId, transactionRequestType: TransactionRequestType,
                                            account : BankAccount, counterparty : BankAccount, body: TransactionRequestBodySandBoxTan,
                                            status: String, charge: TransactionRequestCharge) : Box[TransactionRequest] = {

we are passing TransactionRequestBodySandBoxTan but the function is called createTransactionRequestImpl

- so I guess we would have to rename - createTransactionRequestImpl as well? 

In other words we are abrastracting the different TransactionRequestTypes away to a certain degree - but at some point we need to get specfic.

**Comment by somanole** [comment]
We can close this issue
